Expand use of AppendOnlyVariableWidthData

pettyjamesm opened the pull request:
## Description
Updates additional places in code to use `AppendOnlyVariableWidthData` (introduced in https://github.com/trinodb/trino/pull/25127) to reduce their memory usage as well.

Also adds a special case to `FlatHashStrategyCompiler` for generating `FlatHashStrategy` classes that have less than 500 columns. In those common cases, simpler logic can be used to in the hash method. Similarly, `FlatHashStrategy#valueIdentical` methods are changed to propagate the variableWidthOffset between calls without allocating a `MutableVariableWidthOffset` instance.

<!-- Mark the appropriate option with an (x). Propose a release note if you can.
More info at https://trino.io/development/process#release-note -->
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

dain reviewed (commented):
in the last commit message it might be easier to understand if you say that you are adding a simple fast path when there are less than 500 columns.  I don't think most people understand the multi class chunking hack

pettyjamesm commented:
> in the last commit message it might be easier to understand if you say that you are adding a simple fast path when there are less than 500 columns. I don't think most people understand the multi class chunking hack

Updated the commit message and added a few more inline code equivalent comments to `FlatHashStrategyCompiler`. Thanks for the review!
